tests/collectors/CMPCollector.mocha.js

```diff
@@ -0,0 +1,457 @@
+                const msg = {
+                    type: 'autoconsentDone',
+                    cmp: 'some cmp',
+                    url: 'some url',
+                };
+                collector.selfTestFrame = 1337;
+                commands.splice(0, commands.length);
+                await collector.handleMessage(msg, 1111);
+                assert.strictEqual(commands.length, 1);
+                assert.deepStrictEqual(commands.pop(), ['Runtime.evaluate', {
+                    expression: `autoconsentStandaloneReceiveMessage({ type: "selfTest" })`,
+                    allowUnsafeEvalBlockedByCSP: true,
+                    contextId: 1337,
+                }]);
+            });
+        });
+
+        describe('eval', () => {
+            it('should execute in main world', async () => {
+                /**
+                 * @type {import('@duckduckgo/autoconsent/lib/messages').ContentScriptMessage}
+                 */
+                const msg = {
+                    type: 'eval',
+                    id: 'some id',
+                    code: '1+1==2'
+                };
+                commands.splice(0, commands.length);
+                collector.isolated2pageworld = new Map([[1111, 2222]]);
+                await collector.handleMessage(msg, 1111);
+                assert.strictEqual(commands.length, 2);
+                assert.deepStrictEqual(commands[0], ['Runtime.evaluate', {
+                    expression: msg.code,
+                    returnByValue: true,
+                    allowUnsafeEvalBlockedByCSP: true,
+                    contextId: 2222,
+                }]);
+
+                assert.deepStrictEqual(commands[1], ['Runtime.evaluate', {
+                    expression: 'autoconsentStandaloneReceiveMessage({ id: "some id", type: "evalResp", result: true })',
+                    allowUnsafeEvalBlockedByCSP: true,
+                    contextId: 1111,
+                }]);
+            });
+        });
+    });
+
+    describe('autoconsent cases', () => {
+        let executionContextCreated;
+        /**
+         * @type {{name: string, callback: Function}}
+         */
+        let bindingCalled;
+        beforeEach(async () => {
+            executionContextCreated = listeners.find(a => a.name === 'Runtime.executionContextCreated');
+            assert(executionContextCreated, 'executionContextCreated listener was set');
+            assert(!commands.some(cmd => cmd[0] === 'Runtime.evaluate'), 'no eval commands expected');
+
+            await executionContextCreated.callback({
+                context: {
+                    id: 1,
+                    origin: 'https://example.com/',
+                    auxData: {}
+                }
+            });
+
+            bindingCalled = listeners.find(a => a.name === 'Runtime.bindingCalled');
+            assert(bindingCalled, 'bindingCalled listener was set');
+        });
+
+        it('no CMP', async function() {
+            this.timeout(3000);
```

Comment:
This is making this unit test slow, can we make the timeout shorter? This is 3s because of `maxTimes = 20, interval = 100`, right? Could we make those retries somehow configurable so we can lower that value in testing?

Reply:
yup, will do